rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -2122,6 +2122,8 @@ check_and_attach_condition(
       }
     }
   }
+  // No guard conditions are available. Set the triggered flag of the wait_set to false.
+  wait_set_data->triggered = false;
```

Comment:
The `wait_set_data->triggered` flag is not just about guard conditions.
I think it shall be set to `false` only when all the conditions are unset. i.e. at the very end of this function when returning `false`.

Reply:
Fixed